scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py

```diff
@@ -59,14 +59,37 @@ def deactivate(self):
         self.view.deactivateButton()
 
     def getWorkspaceNames(self):
-        name = self.view.workspace
+        # get current values
+        original_Re_name = self.view.workspace
+        original_Im_name = self.view.imaginary_workspace
         final_options = self.load.get_workspace_names_for_FFT_analysis(self.view.use_raw_data)
 
+        # update view
         self.view.addItems(final_options)
         self.view.removeRe('PhaseQuad')
         self.removePhaseFromIM(final_options)
 
-        self.view.workspace = name
+        # make intelligent guess of what user wants
+        group_pair_context = self.load.group_pair_context[self.load.group_pair_context.selected]
```

Comment:
I think this name is misleading as unless I'm mistaken this is just the currently selected group/pair object?